base/regex.jl

```diff
@@ -154,22 +155,25 @@ r"a.a"
 julia> ismatch(rx, "aba")
 true
 
+julia> ismatch(rx, "aba", 2)
+true
+
 julia> ismatch(rx, "abba")
 false
 
 julia> rx("aba")
 true
 ```
 """
-function ismatch(r::Regex, s::AbstractString, offset::Integer=0)
+function ismatch(r::Regex, s::AbstractString, idx::Integer=start(s))
     compile(r)
-    return PCRE.exec(r.regex, String(s), offset, r.match_options,
+    return PCRE.exec(r.regex, String(s), Csize_t(idx-1), r.match_options,
```

Comment:
`Csize_t` is a C implementation detail, better keep it inside `PCRE.exec` as much as possible. Can't you adapt that function directly?

Reply:
Actually, is any change really needed? `PCRE.exec` uses `ccall`, which will take care of converting `offset` to `Csize_t`. There's no need to make the conversion manually.

Reply:
You are right. I thought, mistakenly understanding the documentation of `ccall`, that the conversion is unsafe, but it is actually safe. Removing.